st2common/st2common/models/db/__init__.py

```diff
@@ -507,9 +507,15 @@ def _process_arg_filters(self, args):
     def _process_null_filters(self, filters):
         result = copy.deepcopy(filters)
 
-        null_filters = {k: v for k, v in six.iteritems(filters)
-                        if v is None or
-                        (type(v) in [str, six.text_type] and str(v.lower()) == 'null')}
+        null_filters = {}
```

Comment:
Python can optimize comprehensions better than for loops, but in this case, readability is probably better. 👍